### PR TITLE
Allow 2nd person conjugation in timer messages

### LINKED ISSUE #5658: MessageRecord_s_set_disappearing_message_time_to_s is bad for 2nd person cases
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

---
### Bug description

Using the following strings for both 2nd and 3rd person cases causes issues:

| string | value |
| --- | --- |
| `MessageRecord_you` | _You_ |
| `MessageRecord_s_set_disappearing_message_time_to_s` | _%1$s set disappearing message time to %2$s_ |

These rely on the assumption that 2nd and 3rd person singular verbs have similar inflections. Many languages don't have this feature.
Even English has some verbs that don't follow this: cf. you **say** vs. cat **says**

I think we should replace `MessageRecord_you` with `MessageRecord_you_set_disappearing_message_time_to_s` ("_You set disappearing message time to %1$s_")

Edit:
Imaginary example in German:
_Du **hat** die Zeit bis zum Verschwinden der Nachricht auf 5 Sekunden festgelegt_ (:exclamation:**FAIL**:exclamation:)
_Franz Biberkopf hat die Zeit bis zum Verschwinden der Nachricht auf 5 Sekunden festgelegt_
### Device info

**Device:** Nexus 4
**Android version:** 6.0.1
**Signal version:** 3.18.1

### PR DESCRIPTION
**Pull request by FeuRenard**
### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
  - Sony Xperia U, Android 4.4.4 (CyanogenMod)
  - AVD Nexus 5X, Android 6.0
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

---
### Description

Fixes #5658
// FREEBIE

### CODE REVIEW

File: src/org/thoughtcrime/securesms/database/model/MessageRecord.java [hunk 1]
```diff
@@ -109,16 +109,26 @@ public SpannableString getDisplayBody() {
...
+      return emphasisAdded(getExpirationTimerUpdateDisplayBody());
     } else if (getBody().getBody().length() > MAX_DISPLAY_LENGTH) {
       return new SpannableString(getBody().getBody().substring(0, MAX_DISPLAY_LENGTH));
     }
 
     return new SpannableString(getBody().getBody());
   }
 
+  private String getExpirationTimerUpdateDisplayBody() {
+    String time   = ExpirationUtil.getExpirationDisplayValue(context, (int)(getExpiresIn() / 1000));
```
Comment (haffenloher) [inline src/org/thoughtcrime/securesms/database/model/MessageRecord.java]: you can remove some spaces before `=` now :)

File: src/org/thoughtcrime/securesms/database/model/MessageRecord.java [hunk 2]
```diff
@@ -109,16 +109,26 @@ public SpannableString getDisplayBody() {
...
+    String time = ExpirationUtil.getExpirationDisplayValue(context, (int)(getExpiresIn() / 1000));
+    String senderAndText;
+    if (isOutgoing()) {
+      senderAndText = context.getString(R.string.MessageRecord_you_set_disappearing_message_time_to_s, time);
+    } else {
+      String sender = getIndividualRecipient().toShortString();
+      senderAndText = context.getString(R.string.MessageRecord_s_set_disappearing_message_time_to_s, sender, time);
+    }
+    return senderAndText;
+  }
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/database/model/MessageRecord.java]: i think this is too verbose

**Comment by FeuRenard** [comment]
I rewrote the method to be less verbose.

File: src/org/thoughtcrime/securesms/database/model/MessageRecord.java [hunk 3]
```diff
@@ -109,16 +109,21 @@ public SpannableString getDisplayBody() {
...
-      return emphasisAdded(context.getString(R.string.MessageRecord_s_set_disappearing_message_time_to_s, sender, time));
+      return emphasisAdded(getExpirationTimerUpdateDisplayBody());
     } else if (getBody().getBody().length() > MAX_DISPLAY_LENGTH) {
       return new SpannableString(getBody().getBody().substring(0, MAX_DISPLAY_LENGTH));
     }
 
     return new SpannableString(getBody().getBody());
   }
 
+  private String getExpirationTimerUpdateDisplayBody() {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/database/model/MessageRecord.java]: please don't break this out into a separate method

**Comment by 2-4601** [comment]
@FeuRenard When you have chance to update this PR don't forget to also add the dot ;) #5764

**Comment by 2-4601** [comment]
> :confused: 

"This branch has conflicts that must be resolved"

**Comment by FeuRenard** [comment]
I saw that too and I am going to resolve this. But it doesn't seem that there is an agreement yet whether to use or don't use the dot in #5764.

**Comment by 2-4601** [comment]
Oh, I thought it was resolved already :smiley: 

https://github.com/WhisperSystems/Signal-Android/issues/5764#issuecomment-252670048
> I found that the messages for generating groups, adding members and changing group names have dots at the end.
>
> So the consequent solution seems in fact to be to add punctuation to these timer change messages.

But I guess Moxie makes the final call but since he hasn't replied anything it's possible he DGAF about the dots :stuck_out_tongue_closed_eyes:

**Comment by moxie0** [comment]
I'm fine either way. Seems like adding a dot to timer messages is the easiest way to be consistent.

**Comment by FeuRenard** [comment]
Fixes #5764

**Comment by moxie0** [comment]
in 3.22.0

**Pull request closed without merging**